RecoTauTag/RecoTau/src/PFRecoTauClusterVariables.cc

```diff
@@ -0,0 +1,367 @@
+#include "RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h"
+
+namespace {
+
+  /// return pf photon candidates that are associated to signal
+  const std::vector<reco::PFCandidatePtr>& getPFGammas(const reco::PFTau& tau, bool signal) {
+    if (signal){
+      return tau.signalPFGammaCands();
+    }
+    return tau.isolationPFGammaCands();
+  }
+  reco::CandidatePtrVector getGammas(const pat::Tau& tau, bool signal) {
+    if(signal){
+      return tau.signalGammaCands();
+    }
+    return tau.isolationGammaCands();
+  }
+  /// decide if photon candidate is inside the cone to be associated to the tau signal
+  bool isInside(float photon_pt, float deta, float dphi) {
+    const double stripEtaAssociationDistance_0p95_p0 = 0.197077;
+    const double stripEtaAssociationDistance_0p95_p1 = 0.658701;
+    const double stripPhiAssociationDistance_0p95_p0 = 0.352476;
+    const double stripPhiAssociationDistance_0p95_p1 = 0.707716;
+    if(photon_pt==0){
+      return false;
+    }      if((dphi<0.3  && dphi<std::max(0.05, stripPhiAssociationDistance_0p95_p0*std::pow(photon_pt, -stripPhiAssociationDistance_0p95_p1))) && (deta<0.15 && deta<std::max(0.05, stripEtaAssociationDistance_0p95_p0*std::pow(photon_pt, -stripEtaAssociationDistance_0p95_p1)))){
+      return true;
+    }
+    return false;
+  }    
+}
+
+namespace reco { namespace tau { namespace mva {
+  /// return chi2 of the leading track ==> deprecated? <==
+  float tau_leadTrackChi2(const reco::PFTau& tau) {
+    float LeadingTracknormalizedChi2 = 0;
+    const reco::PFCandidatePtr& leadingPFCharged = tau.leadPFChargedHadrCand() ;
+    if (leadingPFCharged.isNonnull()) {
+      reco::TrackRef tref = leadingPFCharged -> trackRef();
+      if (tref.isNonnull()) {
+        LeadingTracknormalizedChi2 = (float)(tref -> normalizedChi2());
+      }
+    }
+    return LeadingTracknormalizedChi2;
+  }
+  /// return ratio of energy in ECAL over sum of energy in ECAL and HCAL
+  float tau_Eratio(const reco::PFTau& tau) {
+    std::vector<reco::PFCandidatePtr> constsignal = tau.signalPFCands();
+    float EcalEnInSignalPFCands = 0;
+    float HcalEnInSignalPFCands = 0;
+    typedef std::vector <reco::PFCandidatePtr>::iterator constituents_iterator;
+    for(constituents_iterator it=constsignal.begin(); it != constsignal.end(); ++it) {
+      reco::PFCandidatePtr & icand = *it;
```

Comment:
some modernization of the code can be done, since we have an opportunity with this refactoring:
- range for loops ```for (auto const& icand : constsignal)```

Making the local variable names conform to the CMS style of starting with lower case for locals and data members/methods would be nice as well.